internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -373,11 +373,16 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 			return reconcileComplete{stop: true, dirty: true}
 		}
 
-		// annotate the agent to inform the agent controller to remove the spoke node when the BMH finishes deprovisioning
-		if _, ok := agent.GetAnnotations()[BMH_FINALIZER_NAME]; !ok {
-			setAnnotation(&agent.ObjectMeta, BMH_FINALIZER_NAME, "true")
+		if bmh.Status.Provisioning.State != bmh_v1alpha1.StateDeleting {
```

Comment:
Looking this over again this is awfully specific. And if the following update fails we might not be in this state in the next reconcile.

We should probably come up with something a bit more resilient for this.

Reply:
Oh nevermind, we have a finalizer and according to [their docs](https://github.com/openshift/baremetal-operator/blob/master/docs/baremetalhost-states.md) deleting is a terminal state for the BMH.

This means it will sit in deleting until we remove the finalizer, so this should be fine.

Reply:
NM
https://github.com/openshift/assisted-service/pull/6666/files#diff-86415614c6ed40c56c5ca2b5f751a65746ef5c83b61e6f021a4f5d817e4425baR398